spaceWeatherApi/Utils/DataUtils.cs

```diff
@@ -0,0 +1,225 @@
+                    analysis.Type,
+                    analysis.Speed,
+                    analysis.Latitude,
+                    analysis.Longitude,
+                    analysis.HalfAngle,
+                    analysis.Link
+                })
+            };
+        }
+
+        /// <summary>
+        ///  Find solar events that might have occurred nearly at the same time
+        /// </summary>
+        /// <param name="FLRdata"></param>
+        /// <param name="CMEdata"></param>
+        /// <returns> List of objects</returns>
+
+        public List<object> FindDetailedCorralatedEvents(List<object> FLRdata, List<object> CMEdata)
+        {
+
+            var flareEvents = FLRdata.Cast<FlareEvent>().Where(f => f.BeginTime.HasValue && f.PeakTime.HasValue).ToList();
+            var CMEEvents = CMEdata.Cast<CMEEvent>().Where(c => c.StartTime.HasValue).ToList();
+
+
+            TimeSpan timeVariation = TimeSpan.FromMinutes(5); // 5 minutes for a slightly wider window
+            TimeSpan maxFlareDuration = TimeSpan.FromHours(2); // Maximum flare duration
+
+            var correlatedEvents = flareEvents
+                .SelectMany(flare => CMEEvents
+                    .Where(cme => flare.BeginTime.HasValue && flare.PeakTime.HasValue && cme.StartTime.HasValue &&
+                                  (cme.StartTime.Value >= flare.BeginTime.Value - timeVariation &&
+                                   cme.StartTime.Value <= flare.PeakTime.Value + timeVariation) &&
+                                  (flare.PeakTime.Value - flare.BeginTime.Value) <= maxFlareDuration)
+                    .Select(cme => CreateDetailedCorrelatedEventReport(flare, cme)))
+                .ToList();
+
+            return correlatedEvents;
+        }
+
+        /// <summary>
+        /// Provides a simple object to store correlated solar events
+        /// </summary>
+        public class CorrelatedEvent
+        {
+            public string? FlareClassType { get; set; }
+            public List<float?>? CMESpeeds { get; set; }
+        }
+
+        /// <summary>
+        ///  Find solar events that might have occurred nearly at the same time
+        /// </summary>
+        /// <param name="FLRdata"></param>
+        /// <param name="CMEdata"></param>
+        /// <returns>CorrelatedEvent list</returns>
+        public List<CorrelatedEvent> FindCorrelatedEvents(List<object> FLRdata, List<object> CMEdata)
+        {
+            var flareEvents = FLRdata.Cast<FlareEvent>().ToList();
+            var CMEEvents = CMEdata.Cast<CMEEvent>().ToList();
+
+            TimeSpan timeWindow = TimeSpan.FromMinutes(30);
+
+            var correlatedEvents = flareEvents
+                .SelectMany(flare => CMEEvents
+                    .Where(cme => cme.StartTime.HasValue && flare.BeginTime.HasValue &&
+                                  Math.Abs((cme.StartTime.Value - flare.BeginTime.Value).TotalMinutes) <= timeWindow.TotalMinutes)
```

Comment:
Same feedback as above. Try to make sure these types of conditionals are easy to understand at a glance.